pandas/io/formats/format.py

```diff
@@ -565,315 +583,446 @@ def __init__(
+        self._col_space: ColspaceType
         if col_space is None:
-            self.col_space = {}
+            self._col_space = {}
         elif isinstance(col_space, (int, str)):
-            self.col_space = {"": col_space}
-            self.col_space.update({column: col_space for column in self.frame.columns})
+            self._col_space = {"": col_space}
+            self._col_space.update({column: col_space for column in self.frame.columns})
         elif isinstance(col_space, Mapping):
             for column in col_space.keys():
                 if column not in self.frame.columns and column != "":
                     raise ValueError(
                         f"Col_space is defined for an unknown column: {column}"
                     )
-            self.col_space = col_space
+            self._col_space = col_space
         else:
-            if len(frame.columns) != len(col_space):
+            if len(self.frame.columns) != len(col_space):
                 raise ValueError(
                     f"Col_space length({len(col_space)}) should match "
-                    f"DataFrame number of columns({len(frame.columns)})"
+                    f"DataFrame number of columns({len(self.frame.columns)})"
                 )
-            self.col_space = dict(zip(self.frame.columns, col_space))
+            self._col_space = dict(zip(self.frame.columns, col_space))
+        assert self._col_space is not None
 
-        self.header = header
-        self.index = index
-        self.line_width = line_width
-        self.max_rows = max_rows
-        self.min_rows = min_rows
-        self.max_cols = max_cols
-        self.max_rows_displayed = min(max_rows or len(self.frame), len(self.frame))
-        self.show_dimensions = show_dimensions
-        self.table_id = table_id
-        self.render_links = render_links
+    @property
+    def max_rows_displayed(self) -> int:
+        return min(self.max_rows or len(self.frame), len(self.frame))
 
-        if justify is None:
-            self.justify = get_option("display.colheader_justify")
+    def _is_in_terminal(self) -> bool:
+        """Check if the output is to be shown in terminal."""
+        return bool(self.max_cols == 0 or self.max_rows == 0)
+
+    @property
+    def max_cols_adj(self) -> Optional[int]:
+        """Number of columns fitting the screen."""
+        self._max_cols_adj: Optional[int]
+
+        if hasattr(self, "_max_cols_adj"):
+            return self._max_cols_adj
+
+        if not self._is_in_terminal():
+            self._max_cols_adj = self.max_cols
+            return self._max_cols_adj
+
+        width, _ = get_terminal_size()
+        if self._is_screen_narrow(width):
+            self._max_cols_adj = width
         else:
-            self.justify = justify
+            self._max_cols_adj = self.max_cols
+        return self._max_cols_adj
 
-        self.bold_rows = bold_rows
-        self.escape = escape
+    def _is_screen_narrow(self, max_width) -> bool:
```

Comment:
is this private for a reason?

Reply:
I thought that this method is not going to be used anywhere outside of the class. So, I made this private. I think some other methods also should be private, but I left them as is for now. Unless you have other thoughts on the topic.